File: leetcodes/9. Palindrome Number/isPalindrome.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        # edge case: if x<0, return false
        '''
        time O(log10(N)), space O(1)
        reverse x and created a new integer as revertedNum until x < revertedNum

        return x == revertedNum (even) or x == revertedNum/10 (odd)

        '''
        ###
        if (x < 0 or (x % 10 == 0 and x != 0)):
            return False
        revertedNum = 0
        while (x > revertedNum):
            revertedNum = revertedNum * 10 + (x % 10)
            x = math.floor(x / 10)

        return x == revertedNum or x == math.floor(revertedNum / 10)

# add math.floor to get correct integer for python, it works in leetcode env

File: leetcodes/9. Palindrome Number/isPalindrome.java
public class Solution {
  public boolean isPalindrome(int x) {
    if (x < 0 || (x % 10 == 0 && x != 0))
      return false;
    int reversedNum = 0;
    while (x > reversedNum) {
      reversedNum = reversedNum * 10 + (x % 10);
      // x = (x - x%10)/10;
      x = x / 10;
      // difference between java and js
    }
    // return x == reversedNum || x == (reversedNum - reversedNum % 10)/10;
    return x == reversedNum || x == reversedNum / 10;
  }
}
